torch/nn/functional.py

```diff
@@ -3077,3 +3077,249 @@ def _pad_circular(input, padding):
         input = torch.cat([input[:, :, :, :, -(padding[-5] + padding[-6]):-padding[-5]], input], dim=4)
 
     return input
+
+@weak_script
+def _in_proj(input, weight, bias, start=0, end=None):
```

Comment:
These should not be in functional.py scope. If you need them, move them under "multi_head_attention_forward".

Reply:
Done. also removed TODO as functions are removed in https://github.com/pytorch/pytorch/pull/20177